### PR TITLE
Move `unacceptable_payment_gateway?` to module Account::Gateway

### PR DESCRIPTION
**Pull request by josemigallas**
Small refactor.

### CODE REVIEW

**Comment by jlledom** [comment]
What's the motivation for this change?

**Comment by mayorova** [comment]
@jlledom maybe [this comment](https://github.com/3scale/porta/pull/3167#discussion_r1082388419) :grimacing: 
I think it's a good one - removing duplicate logic.

File: lib/developer_portal/app/controllers/developer_portal/base_controller.rb [hunk 1]
```diff
@@ -21,7 +21,8 @@ def require_credit_card_on_signup?
   end
 
   def finish_signup_for_paid_plan
-    return if unacceptable_payment_gateway?(site_account) || !require_credit_card_on_signup?
```
Comment (akostadinov) [inline lib/developer_portal/app/controllers/developer_portal/base_controller.rb:24]: where was this method defined? I think deginition needs to be removed.

File: test/unit/account/gateway_test.rb [hunk 2]
```diff
@@ -126,4 +126,11 @@ class Account::GatewayTest  < ActiveSupport::TestCase
     buyer.payment_detail.payment_method_id = 'pm_1I5s3n2eZvKYlo2CiO193T69'
     assert_instance_of ActiveMerchant::Billing::StripePaymentIntentsGateway, buyer.provider_payment_gateway
   end
+
+  test 'unacceptable_payment_gateway?' do
+    assert FactoryBot.create(:simple_account, payment_gateway_type: :bogus).unacceptable_payment_gateway?
+    assert FactoryBot.create(:simple_account, payment_gateway_type: nil).unacceptable_payment_gateway?
+
+    assert_not FactoryBot.create(:simple_account, payment_gateway_type: :braintree_blue).unacceptable_payment_gateway?
+  end
```
Comment (akostadinov) [inline test/unit/account/gateway_test.rb:135]: it is a nice refactoring. When I asked in the past about such test though, I thought about where would we actually see something like this. More of an integration test. It is ok to keep as is though if you have more useful other things to do.

**Review by mayorova — APPROVED** [review]
Nice :+1:

File: lib/developer_portal/app/controllers/developer_portal/base_controller.rb (hunk 1 above)
Comment (josemigallas) [reply lib/developer_portal/app/controllers/developer_portal/base_controller.rb:24]: It has been removed: https://github.com/3scale/porta/pull/3171/files/523a027a8c0bd2acabd598745c63785e2b298e1e#diff-9c5e547a8ac256ed8540b94e2432530835017daa884433c71b40ad1e0d4fdcabL21

File: test/unit/account/gateway_test.rb (hunk 2 above)
Comment (josemigallas) [reply test/unit/account/gateway_test.rb:135]: Yeah, this is not supposed to be the test coverage I intended to implement! 😅

**Review by akostadinov — APPROVED** [review]
Cool, my search foo failed me.

**Pull request merged**